images/hook/entrypoint.sh

```diff
@@ -1,44 +1,112 @@
-#!/bin/sh
+#!/bin/bash
 
 set -eu
 
 echo "--> Assuming changeset from the environment: $RIG_CHANGESET"
 
-if [ $1 = "update" ]; then
+function get_control_plane_version() {
+  MAYA_POD=$(kubectl get pod -n openebs | grep -i api | cut -d" " -f1)
+  VERSION=$(kubectl exec -it "${MAYA_POD}" mayactl version -nopenebs | grep ^Version | cut -d" " -f2 | perl -pe's/(\d+.\d+.\d+).*/$1/')
 
-    echo "--> Checking: $RIG_CHANGESET"
-    if rig status ${RIG_CHANGESET} --retry-attempts=1 --retry-period=1s; then
-        exit 0
-    fi
+  echo "$VERSION"
+}
 
-    echo "--> Starting upgrade, changeset: $RIG_CHANGESET"
-    rig cs delete --force -c cs/${RIG_CHANGESET}
+# Verifies that the control plane components are of the expected version
+function check_control_plane() {
+  echo "Checking control plane for version=$1"
 
-    # TODO(r0mant): As this is the first release of OpenEBS as a Gravity application,
-    #               the current "upgrade" procedure assumes that OpenEBS is not
-    #               yet installed and thus just creates all OpenEBS resources
-    #               from scratch.
-    #               In the future, we'll need to develop a proper upgrade procedure
-    #               for OpenEBS components and its pools/volumes like described in
-    #               https://github.com/openebs/openebs/tree/master/k8s/upgrades.
+  CONTROL_PLANE_COMPONENTS=control_plane_components.txt
+  kubectl get pods -n openebs -l openebs.io/version="$1" > $CONTROL_PLANE_COMPONENTS
 
-    echo "--> Creating OpenEBS resources"
-    rig upsert -f /var/lib/gravity/resources/openebs-operator.yaml --debug
+  echo "Found control plane components:"
+  cat $CONTROL_PLANE_COMPONENTS
 
-    echo "--> Checking status"
-    rig status ${RIG_CHANGESET} --retry-attempts=120 --retry-period=1s --debug
+  grep 'provisioner.*Running' $CONTROL_PLANE_COMPONENTS >/dev/null &&
+    grep 'admission-server.*Running' $CONTROL_PLANE_COMPONENTS >/dev/null &&
+    grep 'maya-apiserver.*Running' $CONTROL_PLANE_COMPONENTS >/dev/null &&
+    grep 'ndm.*Running' $CONTROL_PLANE_COMPONENTS >/dev/null &&
+    grep 'snapshot-operator.*Running' $CONTROL_PLANE_COMPONENTS >/dev/null
 
-    echo "--> Freezing"
-    rig freeze
+  return $?
+}
 
-elif [ $1 = "rollback" ]; then
+if [ "$1" = "update" ]; then
+  TO_VERSION=2.2.0
```

Comment:
Instead of hard coding the version 2.2.0 here, I would look for ways to propagate the value from the Makefile. The easiest way I can think of off hand, is probably just to have the Makefile pass a build-arg to docker, and for docker to embed an env variable in the container with TO_VERSION. This way we don't have to hunt for every reference to 2.2.0 if we ever change this in the future.